src/main/java/io/vertx/codegen/type/TypeMirrorFactory.java

```diff
@@ -145,51 +150,34 @@ public TypeInfo create(TypeUse use, DeclaredType type, boolean checkTypeArgs) {
                 handlerArg = create(resolved);
               }
             }
-            raw = new ApiTypeInfo(fqcn, genAnn.concrete(), typeParams, handlerArg, module, nullable, proxyGen);
-          } else if (elt.getAnnotation(DataObject.class) != null) {
-            boolean serializable = Helper.isDataObjectAnnotatedSerializable(elementUtils, elt);
-            boolean deserializable = Helper.isDataObjectAnnotatedDeserializable(elementUtils, typeUtils, elt);
-            TypeInfo jsonType = this.create(elementUtils.getTypeElement("io.vertx.core.json.JsonObject").asType());
-            // Data object annotated here
-            MapperInfo serializer = null;
-            if (serializable) {
-              serializer = new MapperInfo();
-              serializer.setQualifiedName(fqcn);
-              serializer.setKind(MapperKind.SELF);
-              serializer.setTargetType(jsonType);
-            }
-            MapperInfo deserializer = null;
-            if (deserializable) {
-              deserializer = new MapperInfo();
-              deserializer.setQualifiedName(fqcn);
-              deserializer.setKind(MapperKind.SELF);
-              deserializer.setTargetType(jsonType);
-            }
-            raw = new DataObjectTypeInfo(
-              fqcn,
-              module,
-              nullable,
-              typeParams,
-              serializer,
-              deserializer,
-              jsonType
-            );
+            raw = new ApiTypeInfo(fqcn, genAnn.concrete(), typeParams, handlerArg, module, nullable, proxyGen, null);
```

Comment:
Did you forget to parse `DataObjectInfo` here?

Reply:
this is on purpose for now, there is no API yet as we have not defined whether we want to annotate `#VertxGen` with `@DataObject` or use a property declaration file. This will be done after. This only change the internal to make this possible.

Reply:
But, indipendent from the API, `TypeMirrorFactory` already contains a list of types with `serializers`/`deserializers` map. I suppose an eventual API should populate this map, so here you just create the `DataObjectInfo` checking ithese two maps

Reply:
BTW now `TypeMirrorFactory` has serializers/deserializers maps, but it also parses every time `@DataObject` annotation. I would prefer to move the `@DataObject` parsing logic somewhere and register the result straight into these two maps, this change could also simplify the code here. (we could eventually do it later in another PR)

Reply:
that's right for the first comment indeed.

for the second comment, that is not possible because we might process a data object that is coming from a class file (another jar file) and thus the `@DataObject` annotation will not be processed. This is the same reason we need to repeat the `@Mapper` annotation in every project that wants such mapping.

Reply:
Ok so this part is prone to further changes, i see

Reply:
I'll do them soon

Reply:
I updated the code to handle this case with extra tests